### PR TITLE
Adding inference deploy

### PR DESCRIPTION
**Pull request by bigbitbus**
# Description

Adding inference cli deploy container to cloud providers.

Deploy Roboflow Inference to a cloud provider.

If your chosen cloud provider is configured on your terminal, inference 
deploy will automatically use your default credentials. If you have not 
configured your cloud provider, you can do so by following the instructions 
for installing and configuring the cloud provider's CLI tools.

Roboflow inference deploy  will create a new virtual machine 
and deploy the Roboflow Inference container to it.

  Usage examples:
  # Deploy to GCP with CPU
  inference cloud deploy --provider gcp --compute-type cpu
  # Deploy to AWS with GPU
  inference cloud deploy --provider aws --compute-type gpu
  # Stop a deployment (without deleting it, you only pay for storage)
  inference cloud stop <deployment-name>
  # Start a stopped deployment
  inference cloud start <deployment-name>
  # Get status of all deployments
  inference cloud status
  # Undeploy a deployment (delete it)
  inference cloud undeploy <deployment-name>
  # SSH into a deployment
  ssh <deployment-name>

Roboflow inference deploy uses sky (https://github.com/skypilot-org/skypilot) 
to launch a new virtual machine in the cloud provider you specify,
and configures it to run the Roboflow Inference container. If you want
to customize the deployment with a custom sky configuration, 
you can pass a custom config file with the --custom flag, like so:

  inference cloud deploy --custom ./custom.yaml

To see a full list of sky configuration options, visit
https://skypilot.readthedocs.io/en/latest/reference/yaml-spec.html

Roboflow inference cli installs the sky CLI tool for you, so you can
also use the sky cli for more advanced deployment options.

Roboflow inference deploy currently supports AWS and GCP, please 
open an issue at https://github.com/roboflow/inference/issues if 
you would like to see other cloud providers supported.

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

Still testing. Locally

## Any specific deployment considerations

For example, documentation changes, usability, usage/costs, secrets, etc.

## Docs

TBD

### CODE REVIEW

File: inference_cli/lib/cloud_adapter.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+  instance_type: n1-standard-4
+  use_spot: false
+  disk_size: 100
+  disk_tier: medium
+  ports: 9001
+  image_id: projects/ml-images/global/images/c0-deeplearning-common-gpu-v20231209-debian-11
+setup: |
+  docker pull roboflow/roboflow-inference-server-gpu
+run: |
+  docker run -d --gpus all -p 9001:9001 PLACEHOLDER roboflow/roboflow-inference-server-gpu
```
Comment (paulguerrie) [inline inference_cli/lib/cloud_adapter.py:47]: Is this PLACEHOLDER left here intentionally?

File: inference_cli/lib/cloud_adapter.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  cpus: 2+
+  use_spot: false
+  disk_size: 100
+  disk_tier: medium
+  ports: 9001
+  image_id: skypilot:gpu-ubuntu-2004
+setup: |
+  docker pull roboflow/roboflow-inference-server-gpu
+run: |
+  docker run -d --gpus all -p 9001:9001 PLACEHOLDER roboflow/roboflow-inference-server-gpu
```
Comment (paulguerrie) [inline inference_cli/lib/cloud_adapter.py:86]: PLACEHOLDER?

**Review by paulguerrie — APPROVED** [review]
Left a couple questions but looks good to me!

File: inference_cli/lib/cloud_adapter.py (hunk 1 above)
Comment (bigbitbus) [reply inference_cli/lib/cloud_adapter.py:47]: It gets substituted in code

File: inference_cli/lib/cloud_adapter.py (hunk 2 above)
Comment (bigbitbus) [reply inference_cli/lib/cloud_adapter.py:86]: It gets substituted in code

**Comment by bigbitbus** [comment]
> Left a couple questions but looks good to me!

Thanks, I left the responses, yea the "PLACEHOLDER" is replaced when the code runs.

**Pull request merged**